lib/spec/staticcacher.sw-spec.js

```diff
@@ -0,0 +1,181 @@
+'use strict';
+
+importScripts('/base/lib/staticcacher.js');
+
+describe('Static cacher', function() {
+
+  var defaultCacheName = 'offline';
+  var defaultUrls = ['http://www.example.com/', 'http://www.mozilla.org/'];
+  var defaultStaticCacher = new StaticCacher(defaultUrls);
+
+  sinon.stub(self, 'fetch', function() {
+    var response = new Response('content', {});
+    return Promise.resolve(response);
+  });
+
+  afterEach(function() {
+    self.fetch.reset();
+    return clearDefaultCache();
+  });
+
+  function clearDefaultCache() {
+    return caches.open(defaultCacheName).then(function(cache) {
+      return cache.keys().then(function(response) {
+        response.forEach(function(element) {
+          cache.delete(element);
+        });
+      });
+    });
+  }
+
+
+  describe('The constructor', function() {
+
+    it('should accept an array of files', function() {
+      var fileList = ['file1', 'file2'];
+      var cacher = new StaticCacher(fileList);
+      expect(cacher.files).to.deep.equal(fileList);
```

Comment:
I would assert on the constructor not throwing instead of an internal. It's dangerous to rely on internals but it's up to you.